stix/idl/processing/spectrogram/stx_science_data_lightcurve.pro

```diff
@@ -0,0 +1,225 @@
+  endif else begin
+    stx_convert_spectrogram, fits_path_data = fits_path, fits_path_bk =  fits_path_bk, distance = distance, time_shift = time_shift, ospex_obj = ospex_obj, _extra= _extra
+  endelse
+
+  data_obj = ospex_obj->get(/obj,class='spex_data')
+  counts_str = ospex_obj->getdata(spex_units='counts')
+
+  ut_time = ospex_obj->getaxis(/ut, /edges_1)
+  ut2_time = ospex_obj->getaxis(/ut, /edges_2)
+  duration = ospex_obj->getaxis(/ut, /width)
+
+  ;use OSPEX bin_data method to bin counts in given energy bands
+  energy_summed_counts = data_obj->bin_data(data = counts_str, intervals = energy_ranges, $
+    eresult = energy_summed_error, ltime = energy_summed_error)
+
+  energy_summed_str = {data:energy_summed_counts, edata:energy_summed_error, ltime:energy_summed_error}
+
+  ; determine time bins with minimum duration - keep adding consecutive bins until the minimum
+  ; value is at least reached
+  i=0
+  j=0
+  total=duration[0]
+  iall=[]
+
+  while (i lt n_elements(duration)-1) do begin
+    while (total lt time_min)  and (i+j le n_elements(duration)-1) do begin
+      total = total(duration[i:i+j])
+      j++
+    endwhile
+    iall = [iall,i]
+    i = i+j
+    j = 0
+    total = 0
+  endwhile
+
+  intervals = [ut2_time[0,iall[0:-2]],ut2_time[1,iall[1:-1]-1]]
+  intervals = [[intervals],[ ut2_time[0,iall[-1]], ut2_time[1,-1]]]
+
+  ;use OSPEX bin_data method to bin counts in given time bands
+  time_summed_counts = data_obj->bin_data(data = energy_summed_str, intervals = intervals, /do_time, $
+    eresult = time_summed_error, ltime = time_summed_livetime)
+
+  ; insert the data summed in time and energy back into the OSPEX object
+  ospex_obj->set, spex_data_source = 'spex_user_data'
+  ospex_obj->set, spectrum = time_summed_counts,  $
+    spex_ct_edges = energy_ranges, $
+    errors = time_summed_error, $
+    spex_ut_edges = intervals, $
+    livetime = time_summed_livetime, $
+    spex_detectors = 'STIX'
+  origunits = ospex_obj->get(/spex_data_origunits)
+  origunits.data_name = 'STIX'
+  ospex_obj->set, spex_data_origunits = origunits
+  ospex_obj->set, spex_uncert = 0.05
+  ospex_obj -> set, spex_eband = energy_ranges
+  ospex_obj -> set, spex_tband = intervals
+
+  ; if the plot_obj keyword is present create a plotman window and plot the lightcurve
+  if arg_present(plot_obj) then begin
+    ospex_obj->plot_time,  spex_units='flux', /show_err, /show_filter
+    pobj = ospex_obj->get_plotman_obj()
+    ut_plot = pobj->get(/data)
+    plot_obj = plotman(desc='STIX Lightcurve', input = ut_plot, /ylog, xst = 1)
+  endif
+
+  ;retrieve the data from the OSPEX object for the output structure
+  flux_str = ospex_obj->getdata(spex_units='flux')
+  ut2_time = ospex_obj->getaxis(/ut, /edges_2)
+  duration = ospex_obj->getaxis(/ut, /width)
+
+  light_curve_str = {$
+    data:flux_str.data, $
```

Comment:
Maybe `counts` to indicate it not rate/livetime corrected. Also what the unit `counts/keV*cm^2` ?

Reply:
Good point. I've updated to include the units. In the future this should be user selectable but right now the type is hard coded to flux which I think is the most generally meaningful.